### ISSUE TITLE
[DEPR]: problem-builder

### ISSUE THREAD

**Issue by xitij2000** [openedx/public-engineering#268]
### Proposal Date

2024-06-25

### Target Ticket Acceptance Date

2024-09-15

### Earliest Open edX Named Release Without This Functionality

Teak - 2025-07-10

### Rationale

Problem Builder and Step Builder were originally sponsored by major clients of OpenCraft, however they are no longer being used or sponsored and as such have not seen significant development in over four years. OpenCraft has merely kept it running with the latest Open edX releases. 

As explained in greater detail [here](https://discuss.openedx.org/t/future-directions-for-problem-builder/10337) the block uses an technical architecture of the core platform (child blocks) that might itself be deprecated and removed in the future, and Problem Builder would need a major refactor in that case. 

OpenCraft doesn't have the resources to continue maintaining it anymore. 

### Removal

The [problem-builder repository](https://github.com/open-craft/problem-builder/) will be marked as deprecated after this is accepted and  eventually archived.

### Replacement

Some of its features are unique but others are available in other blocks or in the core platform. 

### Migration

We can help create tools and scripts for migration if there is community interest and support. 

### Additional Info

OpenCraft are also open to help integrate some of its functionality into the core platform, or handing over the ownership of the block to an interested party.

**Comment by xitij2000** [comment]
We've decided to look into the possibility of creating a simplified version of the problem-builder block that retains some core features while removing lesser-used features. We'd hope to use this as a proving ground for the [XBlock v2 proposal](https://docs.google.com/document/d/1xelb7iGvWRtT71F2c_dBkQEB47Cj0etT5VGgbVdAPKY/edit) that Braden shared during the conference and help strengthen that approach by using a real-world example.

**Comment by feanil** [comment]
@xitij2000 checking back in on this, do you still want to pursue this DEPR?

**Comment by robrap** [comment]
[inform/request] This ticket was accepted under the old DEPR process. The [updated Process States in OEP-21](https://open-edx-proposals.readthedocs.io/en/latest/processes/oep-0021-proc-deprecation.html#process-states) help add clarity to communications and timing regarding the introduction of breaking changes. This ticket must move to Transition Unblocked before moving to Breaking Changes Unblocked. The default minimum time to wait in Transition Unblocked is 1 month, so early deployers have time to prepare. If you wish to make breaking changes in less than 1 month, please initiate discussions to negotiate a date as soon as possible. Thank you!

**Comment by xitij2000** [comment]
@feanil @bradenmacdonald, I and others at OpenCraft been having some discussions internally about using this block as a playground for testing XBlock v2 so we've deferred the deprecation till we work on that. 

@robrap This ticket is about a year old, was [announced on the forums](https://discuss.openedx.org/t/future-directions-for-problem-builder/10337) almost about 9 months ago and the [initial discussions](https://discuss.openedx.org/t/future-directions-for-problem-builder/10337) on the forums started well before that. The above XBlock v2 approach would allow transitioning to a partial replacement, however, even without a replacement I think by this point of time it would be fair to move it to "Breaking Changes Unblocked" or do we need to reset the ticket flow here?

**Comment by feanil** [comment]
@xitij2000 it sounds like maybe the best option is to deprecate this repo and then in the future the repo may come back to life as a v2 XBlock but since it's not installed by default, we can finish the deprecation of the current version of the block so that people know that it is not going to get more support in its current iteration.

As I said, this block is not currently installed by default and so from a maintenance perspective, it's pretty easy to manage the deprecation.

I recommend the following:

* Update the repo to indicate that it is no longer guaranteeing updates and is not gauranteed to work with newer release of the Open edX Platform.
* Re-announce the deprecation since it's been a while when this ticket was in limbo.
* Close this ticket
* Continue to iterate on that repo as you like to experiment with Xblock V2.

Let me know what you think of that.

**Comment by xitij2000** [comment]
@feanil Sorry for the late response. I think OpenCraft plan to maintain it till the replacement is ready, I will check on this and get back to you. 

Since the new block will likely not be a 100%  replacement your proposed line of action does make sense.

**Comment by feanil** [comment]
Great, in that case I think it's safe to close this ticket for now.  Let me know or re-open it if you disagree.
